src/utils/utils.ts

```diff
@@ -69,3 +69,83 @@ export const createDatesArray = (
   }
   return datesArray
 }
+
+interface CleanedEarn {
+  _id: number
+  data: [
+    {
+      minFee: number
+      maxFee: number
+      dayCount: number
+      memCount: number
+      minDelay: number
+      maxDelay: number
+      minMinutes: number
+      maxMinutes: number
+    }
+  ]
+}
+
+interface EarnAllInfo {
+  _id: number
+  data: [
+    {
+      minFee: number
+      maxFee: number
+      dayCount: number
+      memCount: number
+      minDelay: number
+      maxDelay: number
+      minMinutes: number
+      maxMinutes: number
+    }
+  ]
+  sortedData: {
+    zeroToOne: number
+    oneToTwo: number
+    twoToThree: number
+    threeToTen: number
+  }
+}
+
+export const earnDataParser = (
+  preParsedData: CleanedEarn
+): EarnAllInfo | Error => {
+  let zeroToOne
+  let oneToTwo
+  let twoToThree
+  let threeToTen
```

Comment:
This can be reduced to one line. Change to `let zeroToOne, oneToTwo, twoToThree, threeToTen`